### PR TITLE
move printf test

### PR DESCRIPTION
**Pull request by mateusz-bloch**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: printf/stdio_printf.c [hunk 1]
```diff
@@ -0,0 +1,2645 @@
...
+					close(fd); \
+					close(stdout_fd); \
+					break; \
+				} \
+				case SNTEST: \
+					snprintf(buff, strlen(expect) + 1, format, value); \
+					break; \
+				case STEST: \
+					sprintf(buff, format, value); \
+					break; \
```
Comment (damianloew) [inline printf/stdio_printf.c:88]: We use braces even for one line blocks.

```suggestion
				case SNTEST: \
					snprintf(buff, strlen(expect) + 1, format, value); \
					break; \
				} \
				case STEST: { \
					sprintf(buff, format, value); \
					break; \
				} \
```

File: printf/stdio_printf.c [hunk 2]
```diff
@@ -0,0 +1,2645 @@
...
+#define NUM_OUTPUT_TYPES 5
+
+#define PATH "stdio_printf_test"
+
+/* comma delimiter macro ',' for multiple arguments parsing to *printf() function via macros */
+#define COMMA ,
+
+/* macro assertions*/
+
+#define printf_test(format, value, expect) \
```
Comment (damianloew) [inline printf/stdio_printf.c:60]: Let's have a common convention in our test and prefix all global functions/variables by `test` and use camelCase as it's recommended in our coding convention.

```suggestion
static FILE *test_outFile;
static char test_buff[700];

enum test_outType { DTEST,
	FTEST,
	TEST,
	SNTEST,
	STEST };

#define NUM_OUTPUT_TYPES 5

#define PATH "stdio_printf_test"

/* comma delimiter macro ',' for multiple arguments parsing to *printf() function via macros */
#define COMMA ,

/* macro assertions*/

#define test_assertPrintfs(format, value, expect) \
```

File: printf/stdio_printf.c [hunk 3]
```diff
@@ -0,0 +1,2645 @@
...
+			} \
+			fseek(output_file, 0, SEEK_SET); \
+			fflush(output_file); \
+			TEST_ASSERT_EQUAL_STRING(expect, fgets(buff, strlen(expect) + 1, output_file)); \
+			TEST_ASSERT_EQUAL_STRING(expect, buff); \
+		} \
+	}
+
+/* wrappers for vprintf tests*/
+static void vprintf_test(char *expect, const char *format, ...)
```
Comment (damianloew) [inline printf/stdio_printf.c:98]:
```suggestion
static void test_assertVprintfs(char *expect, const char *format, ...)
```

**Review by damianloew — CHANGES_REQUESTED** [review]
Thanks for decreasing the size of binary. We can order names of variables/functions by an occasion. Above that please prefix commit messages by `libc_`, give some commit description about why decreasing size/moving to the separate binary is needed. Please place this directory inside the libc one as you suggested.

File: printf/stdio_printf.c (hunk 1 above)
Comment (nalajcie) [reply printf/stdio_printf.c]: please note that adding subscopes (`{ }`) in switch `case's` is not needed (unless You're declaring variables there) and IMHO decreases readability.

Comment (damianloew) [reply printf/stdio_printf.c]: Yeah right, it isn't needed there, we will remove them.

File: libc/printf/test.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+test:
+  type: unity
+  tests:
+    - name: unit
```
Comment (damianloew) [inline libc/printf/test.yaml:4]: We always want to leave a comment, why do we use an unusual approach, like this one:
```suggestion
    # Executing test-printf on 'armv7m4-stm32l4x6-nucleo' takes ~9 min, that's why we want to run it only during nightly campaigns 
    - name: unit
```

**Review by damianloew — APPROVED** [review]
Thanks for the changes, it looks much simpler

**Pull request merged**